programs/bench.c

```diff
@@ -213,317 +431,251 @@ BMK_return_t BMK_benchMem(const void* srcBuffer, size_t srcSize,
     U32 nbBlocks;
 
     /* checks */
-    if (!compressedBuffer || !resultBuffer || !blockTable)
-        EXM_THROW(31, "allocation error : not enough memory");
+    if (!compressedBuffer || !resultBuffer ||  
+        !srcPtrs || !srcSizes || !cPtrs || !cSizes || !resPtrs || !resSizes)
+        EXM_THROW(31, BMK_return_t, "allocation error : not enough memory");
```

Comment:
OK, we have a new problem here.

Previously, `EXM_THROW` was exiting the program.
As a consequence, any allocated memory would be freed there just while mopping up the process.

But now, `EXM_THROW` just returns, 
which is a significantly different behavior (likely worth a renaming).
As consequence, if 6 buffers are allocated but the 7th fail, we have a memory leak on exiting the function here.

Make sure all buffers allocated in this function are freed when exiting the function.

This can be difficult when there are multiple exit points, as you'll have to free all resources at each and every exit point. That's a significant burden.

A strategy for this situation is to group all exit responsibilities in one place.
One I like is to "nest" functions into 2 levels : 

```
int function1(...)
{
   ... Allocate stuff ...
   if (allocation_successful)
      result = function2(...)
   ... Free stuff ...
   return result;
}

int function2(...)
{
   /* do not allocate anything here */
   ... do stuff...
   ... exit 1 ...
   ... do more stuff ...
   ... exit 2 ...
   ... do more stuff ...
   ... final exit ...
}
```

In some cases, the amount of variables to pass from function1 to function2 can be too large to feel comfortable.
A second possibility then is to use a `goto` to an exit section.